### PR TITLE
Filters

### PR DESCRIPTION
**Pull request by Sameony**
Filter functionality in the product display page

https://user-images.githubusercontent.com/80711125/161596685-512a12a4-8970-436f-b173-185e55f8f67e.mp4

### CODE REVIEW

File: src/backend/db/products.js [hunk 1]
```diff
@@ -10,30 +10,192 @@ export const products = [
...
+    color: "brown",
+    prevPrice:"18999",
+    price: 18000,
+    reviews: "3439",
+    rating: 4.6,
+    categoryName: "pattern-blankets",
+  },
+  {
+    _id: uuid(),
+    imageUrl: "Images/pawPlainBlanky.png",
```
Comment (shuntzhere) [inline src/backend/db/products.js:23]: You can add the images here directly. No need to create extra folder for it.

**Review by shuntzhere — APPROVED** [review]
The UI and the code look good, would suggest making a better folder structure. Currently it looks confusing.

File: src/components/filters.jsx [hunk 2]
```diff
@@ -4,10 +4,10 @@ import { useFilters } from '../context/filterContext';
 const Filters = () => {
     const { state, filterDispatch } = useFilters();
   return (
-    <aside className="filters">
+    <form className="filters">
```
Comment (vaibhav-mahalle) [inline src/components/filters.jsx]: any specific reason to use form instead of aside?

File: src/components/filters.jsx [hunk 3]
```diff
@@ -1,52 +1,50 @@
...
+                onChange={(e) =>filterDispatch({ type: "MAXPRICE", payload: e.target.value }) }/>
+                
+                <p className='mg-l-sm'>MaxPrice: <span className='snippets mg-l-sm'>{state.maxPrice}</span></p>
             </div>
             <div className="mg-y-md">
                 <h3>Category</h3>
                 <input type="checkbox" className="mg-sm" name="shopping" id="plainBlankets" value="plain-blankets" checked={state.category.includes("plain-blankets")}
                 onChange={(e) =>filterDispatch({ type: "CATEGORY", payload: e.target.value })}/>
-                <label for="plainBlankets">Plain-Blankets</label> <br />
+                <label htmlFor="plainBlankets">Plain-Blankets</label> <br />
```
Comment (vaibhav-mahalle) [inline src/components/filters.jsx:22]: is it necessary to use htmlFor here ?

**Review by vaibhav-mahalle — APPROVED** [review]
learnt quite a few things while reviewing this pr. Have raised some doubts please look into them. Keep up the good practice of naming good commit messages.

File: src/components/filters.jsx (hunk 3 above)
Comment (Sameony) [reply src/components/filters.jsx:22]: yes, react otherwise throws a warning

File: src/backend/db/products.js (hunk 1 above)
Comment (Sameony) [reply src/backend/db/products.js:23]: oh, alright, will see to it
thanks

Comment (Sameony) [reply src/backend/db/products.js:23]: as for the images part, i have kept them in public thats why i am supplying that path in the db itself

File: src/components/filters.jsx (hunk 2 above)
Comment (Sameony) [reply src/components/filters.jsx]: was trying the reset button type for "clear" in filters...switched back to <aside> in latest patch

**Pull request merged**